tests/test_utils.py

```diff
@@ -0,0 +1,23 @@
+# This file is part of REANA.
+# Copyright (C) 2021 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA-Server tests for utils module."""
+
+import pytest
+from reana_server.utils import is_valid_email
+
+
+@pytest.mark.parametrize(
+    "email,is_valid",
+    [
+        ("john.doe@example.org", True),
+        ("john-doe@example.org", True),
```

Comment:
A good test persona is `jean-yves.le.meur@cern.ch`. It should pass.

Reply:
Would be nice to support also emails like `john.doe.poe@example.org` (which you found that it's not working). While testing it [here](https://regex101.com/r/S9bgjX/1) it seems it's not working currently

Reply:
This will not pass rn. We only accept a single dot before `@`. More details in PR description. We have two options: (A) to modify the current regex to allow two (or more?) dots or (B) use regex I found online which is less strict (allows for some incorrect values) but covers a lot of cases.

What should be the way?

Reply:
I am inclined towards a less strict regex but I don't know how important strictness is for us.

Reply:
W3C email regex used in type=email HTML field ([details](https://html.spec.whatwg.org/multipage/input.html#email-state-(type=email))) :) 

```python
email_regex = r"^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$"
```

Reply:
Yes, I'm also for having more permissive check, so that emails like that of Jean-Yves would pass.  (That was what I meant by the terse expression "it should pass"  there -- i.e. I have tested that one and it didn't. So it's nice to add it in a TDD style and then change the implementation to make it pass...)

Reply:
I changed to the regex that I posted in the PR description. Now tests are passing. The new one is more permissive but also allows for some incorrect values (e.g `john@example.org..`). I think it is good for now. In case admins will want to have a stricter check we can reiterate.

I didn't want to use the W3C regex as it is pretty huge and can be intimidating. It covers *some* edge cases (e.g `john@example.org..`) but we are fine without it.

Reply:
Well if it's battle-tested, I'd say let's go for the W3C one! Better to catch trailing dots...

Reply:
(e.g. the current one would accept happily `john@example....org` which is not good... while the W3C one is working better. Perhaps add all these double-dots into test suite so that we retain easily what we tested...)

Reply:
Done. 

Interesting thing is that `john@exampleorg` is a valid email. HTML input (type=email) allows it.

Reply:
It is definitely nicer to use the Marshmallow one :+1:

Reply:
Agree. It is already installed for `r-server`. I will change PR.

Reply:
Done